opentreemap/treemap/js/src/mapPage/locationSearchUI.js

```diff
@@ -41,11 +51,19 @@ function clearLocationInput() {
     showAppropriateWellButton();
 }
 
-function onSearchChanged() {
-    var text = $(dom.locationInput).val();
+function onSearchChanged(searchEvent) {
+    var text = $(dom.locationInput).val(),
+        boundaryId = (!!searchEvent && !!searchEvent.filter &&
+                      !!searchEvent.filter &&
+                      !!searchEvent.filter['mapFeature.geom'] &&
+                      searchEvent.filter['mapFeature.geom'].IN_BOUNDARY) ||
+                      null;
     if (text) {
-        showControls(dom.controls.searched);
         $(dom.locationSearched).html(text);
+        showControls(dom.controls.searched);
+    } else if (boundaryId !== null) {
+        $(dom.locationValue).val(String(boundaryId));
+        showControls(dom.controls.customArea);
```

Comment:
Currently there's a bug -- if you search for a custom boundary and then reload the page, we think you have a custom area. But the logic here should suffice, since there should be `text` in the custom boundary case.

I suspect it's a sequencing issue -- that this code is called before `Search.applyFilterObjectToDom`.